tests/unit/test_database.py

```diff
@@ -2679,6 +2693,163 @@ def test_process_w_query_batch(self):
         )
 
 
+class TestMutationGroupsCheckout(_BaseTest):
+    def _get_target_class(self):
+        from google.cloud.spanner_v1.database import MutationGroupsCheckout
+
+        return MutationGroupsCheckout
+
+    @staticmethod
+    def _make_spanner_client():
+        from google.cloud.spanner_v1 import SpannerClient
+
+        return mock.create_autospec(SpannerClient)
+
+    def test_ctor(self):
+        from google.cloud.spanner_v1.batch import MutationGroups
+
+        database = _Database(self.DATABASE_NAME)
+        pool = database._pool = _Pool()
+        session = _Session(database)
+        pool.put(session)
+        checkout = self._make_one(database)
+        self.assertIs(checkout._database, database)
+
+        with checkout as groups:
+            self.assertIsNone(pool._session)
+            self.assertIsInstance(groups, MutationGroups)
+            self.assertIs(groups._session, session)
+
+        self.assertIs(pool._session, session)
+
+    def test_context_mgr_success(self):
+        import datetime
+        from google.cloud.spanner_v1 import BatchWriteRequest
+        from google.cloud.spanner_v1 import BatchWriteResponse
+        from google.cloud._helpers import UTC
+        from google.cloud._helpers import _datetime_to_pb_timestamp
+        from google.cloud.spanner_v1.batch import MutationGroups
+        from google.rpc.status_pb2 import Status
+
+        now = datetime.datetime.utcnow().replace(tzinfo=UTC)
+        now_pb = _datetime_to_pb_timestamp(now)
+        status_pb = Status(code=200)
+        response = BatchWriteResponse(
+            commit_timestamp=now_pb, indexes=[0], status=status_pb
```

Comment:
I don't understand this case.
In the response we have indexes=[0], whereas request passes empty `mutation_groups`.

Reply:
Agree that the test checks only the expected request parameters. But mocking correct response would help someone looking into this code later and not get confused.

Reply:
Updated.